### DISCUSSION TITLE
Workspace in GRBL does not match workspace of engraver, engraves backwards and opposite sides

### DISCUSSION THREAD

**Discussion by The-LoneStar** [arkypita/LaserGRBL#1798] [Q&A]
I'm very new to laser engraving and GRBL. I REALLY hope someone can help me with my issue. My laser is a Wainlux L3 that just came out and has a workspace of approximately 300mm x 300mm by my measurement. The engraver only uses maybe a third of that area. When I try to engrave in the whole area I get an error message that I'm out of range of my engraver. How do I fix this? I also have another weird issue. When I try to engrave at the top it engraves at the bottom! Same thing with right and left. It also engraves images backwards? How do I fix this? I appreciate any help with these issues that I have. Thank you.
![grbl_code](https://user-images.githubusercontent.com/102552825/160523224-169450da-a168-4fe5-b324-1e82fb130154.png)

**Comment by StuartB4** [comment]
Start LaserGRBL and click connect, goto Grbl configuration on the Grbl menu. 
Change the ones I've highlighted to the same numbers in the Value boxes and click Write and Close. 
You can leave $30 and 32 though, thats already correct. 

![123123](https://user-images.githubusercontent.com/16980026/160524822-4d0c261f-fe75-4783-a824-d9a70a101f73.png)

**Comment by The-LoneStar** [comment]
Thank you so much for helping me, I really appreciate it. I made the
changes so now when it homes it goes to the upper right hand corner but
just sits there and grinds trying to keep moving. It appears I have a limit
switch on the machine. How do I fix this problem? Thank you.

**Comment by Tribo-2dot0** [comment]
Try changing $23=3
![FFXZ8LSKCQ8IADI](https://user-images.githubusercontent.com/83982525/160669348-7477760e-b7bb-4045-b281-b3879743b92a.jpg)

**Comment by Viaczeslav** [comment]
Hello Friend.I  need firmware  from WAINLUX to update MAINBOARD to GRBL.Very much please share.Thank you
